### PR TITLE
feat: github workflow for sdk package testing

### PR DESCRIPTION
**Pull request by typotter**
🎟️ towards: FF-3104

## Motivation and context
The next link in the chain is to run the SDK production-level testing on github.

## Changes 
- new reusable workflow file to run the test cluster against an SDK server relay
- test runner tweak to prefer the non-docker runner.

### CODE REVIEW

File: .github/workflows/test-sdk-packages.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Test Packaged SDKs
+
+on:
+  pull_request:
+    paths:
+      - ufc/**
```
Comment (typotter) [inline .github/workflows/test-sdk-packages.yml:6]: run the tests on in-flight pull reviews that change the UFC test data

File: package-testing/sdk-test-runner/test-sdk.sh [hunk 2]
```diff
@@ -136,20 +136,23 @@ case "$command" in
         pushd ../$SDK_DIR
 
         BUILD_AND_RUN_PLATFORM=build-and-run-${EPPO_SDK_PLATFORM}.sh
-        if [ -f docker-run.sh ]; then
+
+        if [ -f build-and-run.sh ]; then
```
Comment (typotter) [inline package-testing/sdk-test-runner/test-sdk.sh:140]: prefer the build-and-run script

File: .github/workflows/test-sdk-packages.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+name: Test Packaged SDKs
```
Comment (leoromanovsky) [inline .github/workflows/test-sdk-packages.yml:1]: ❤️ ❤️ ❤️

File: .github/workflows/test-sdk-packages.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  test-php-sdk:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ['linux']
+    uses: ./.github/workflows/test-server-sdk.yml
+    with:
+      platform: ${{ matrix.platform }}
+      sdkName: 'eppo/php-sdk'
```
Comment (leoromanovsky) [inline .github/workflows/test-sdk-packages.yml:22]: very nice - my understanding of how this fits together is the `test-server-sdk.yml` is templatized and we will add additional SDKs here as they become available

**Review by leoromanovsky — APPROVED** [review]

**Review by aarsilv — APPROVED** [review]
📈

File: .github/workflows/test-sdk-packages.yml (hunk 4 above)
Comment (typotter) [reply .github/workflows/test-sdk-packages.yml:22]: @leoromanovsky you got it!
I tried to configure it with the SDK as a dimension in the runner matrix, but each language needs to test against different dimensions and the job matrix doesn't work like that.

**Pull request merged**